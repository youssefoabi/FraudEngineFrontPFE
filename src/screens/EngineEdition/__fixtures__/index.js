export default {
  id: 1,
  code: 0,
  name: 'name',
  isEnable: true,
  scopes: [
    {
      scopeId: 1,
      name: "gamme active avec un param n'est pas lie...",
      order: 0,
      isEnable: true,
    },
    {
      scopeId: 2,
      name: 'gamme active valide',
      order: 0,
      isEnable: true,
    },
    {
      scopeId: 3,
      name: 'gamme disactive',
      order: 0,
      isEnable: false,
    },
  ],
  inputFields: [
    {
      id: 0,
      type: 'Address',
      order: 0,
      label: 'Placeholder Field 1',
      isEnable: true,
      isMandatory: true,
      parameters: [
        {
          id: 4,
          scopeId: 2,
          key: 'Latitude',
          externalCodeId: 4,
        },
      ],
    },
    {
      id: 1,
      type: 'Address',
      order: 0,
      label: 'Placeholder Field 2',
      isEnable: true,
      isMandatory: true,
      parameters: [
        {
          id: 0,
          scopeId: 1,
          key: 'cp',
          externalCodeId: 2,
        },
        {
          id: 1,
          scopeId: 1,
          key: 'commune',
          externalCodeId: 4,
        },
        {
          id: 3,
          scopeId: 2,
          key: 'search',
          externalCodeId: 1,
        },
      ],
    },
  ],
  backgroundImages: [
    {
      order: 0,
      urlImageDesktop:
        'https://www.leparisien.fr/resizer/CBn4GjacGJ5llevItKUbQYa6zcw=/932x582/arc-anglerfish-eu-central-1-prod-leparisien.s3.amazonaws.com/public/4U4PGCRHAK2F6HLMZCBI4PSDNU.jpg',
      urlLinkDesktop: 'Lien PC',
      urlImageMobile:
        'https://www.leparisien.fr/resizer/CBn4GjacGJ5llevItKUbQYa6zcw=/932x582/arc-anglerfish-eu-central-1-prod-leparisien.s3.amazonaws.com/public/4U4PGCRHAK2F6HLMZCBI4PSDNU.jpg',
      urlLinkMobile: 'Lien Mobile',
      alt: 'alternative text',
      openInNewTab: true,
      isEnable: true,
    },
  ],
  logo: {
    urlImageDesktop: 'string',
    urlLinkDesktop: 'string',
    urlImageMobile: 'string',
    urlLinkMobile: 'string',
    alt: 'string',
    openInNewTab: true,
    isEnable: true,
  },
  marketingText: {
    text: 'marketingText',
    isEnable: true,
  },
  searchText: 'searchText',
  createdAt: '2020-05-13T15:39:07.932Z',
  modifiedAt: '2020-05-13T15:39:07.932Z',
};
