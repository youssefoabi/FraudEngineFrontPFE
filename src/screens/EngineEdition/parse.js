import { addIndex, find, fromPairs, isNil, map, pathOr, pipe, prop } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

import { typesOptions } from '../../components/SearchCriteriaSettings/components/FieldsBlock/config';

const toTimestamp = date => (date ? Number(new Date(date)) : null);

const parseEngine = engine => {
  if (isNil(engine)) return null;

  const {
    id,
    code,
    name,
    isEnable,
    scopes,
    inputFields,
    backgroundImages,
    marketingText,
    searchText,
    createdAt,
    modifiedAt,
  } = engine;

  const { text, isEnable: isMarketingTextEnable } = marketingText || {};

  const mapScopes = allScopes => {
    if (isNilOrEmpty(allScopes)) return null;

    return pipe(
      map(scope => [prop('scopeId', scope), scope]),
      fromPairs,
    )(allScopes);
  };

  const mapInputFields = allInputFields => {
    if (isNilOrEmpty(allInputFields)) return null;
    const mapIndexed = addIndex(map);

    return pipe(
      map(inputField => {
        const type = pathOr(null, ['type'], inputField);
        return { ...(inputField || {}), type: find(option => type === prop('value', option), typesOptions) };
      }),
      mapIndexed((inputField, index) => [index, inputField]),
      fromPairs,
    )(allInputFields);
  };

  const mapBackgroundImages = allBackgroundImages => {
    if (isNilOrEmpty(allBackgroundImages)) return null;

    const mapIndexed = addIndex(map);
    return pipe(
      mapIndexed((backgroundImage, index) => [index, backgroundImage]),
      fromPairs,
    )(allBackgroundImages);
  };

  const firstFieldId = pathOr(null, ['0', 'id'], inputFields);
  const secondFieldId = pathOr(null, ['1', 'id'], inputFields);

  return {
    id: code,
    engineId: id,
    name,
    isEnable,
    scopes: mapScopes(scopes),
    inputFields: mapInputFields(inputFields),
    backgroundImages: mapBackgroundImages(backgroundImages),
    marketingText: text,
    isMarketingTextEnable,
    searchText,
    createdAt,
    modifiedAt: toTimestamp(modifiedAt),
    firstFieldId,
    secondFieldId,
  };
};
export default parseEngine;
