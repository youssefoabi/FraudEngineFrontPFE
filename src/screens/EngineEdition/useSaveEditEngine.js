import { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { EDIT_ENGINE_REQUEST } from '../../components/EngineForm/actions';

export default function useSaveEditEngine() {
  const dispatch = useDispatch();

  const saveEditEngine = useCallback(engine => dispatch({ type: EDIT_ENGINE_REQUEST, payload: { engine } }), [
    dispatch,
  ]);

  return { saveEditEngine };
}
