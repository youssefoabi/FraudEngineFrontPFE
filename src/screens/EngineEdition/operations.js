import { call, put, takeLatest } from 'redux-saga/effects';

import {
  EDIT_ENGINE_ERROR,
  EDIT_ENGINE_REQUEST,
  EDIT_ENGINE_SUCCESS,
} from '../../components/EngineForm/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';
import parse from '../EngineCreation/parse';

export function* saveEngine({ payload }) {
  yield put({ type: PENDING_REQUEST });

  try {
    const { engine } = payload;
    if (!engine) return null;

    yield call(clientApi().updateEngine, parse(engine));
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: EDIT_ENGINE_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: EDIT_ENGINE_ERROR,
      error: { errorMessage: "Erreur d'enregistrement de moteur de recherche" },
    });
  }

  return true;
}

export default [takeLatest(EDIT_ENGINE_REQUEST, saveEngine)];
