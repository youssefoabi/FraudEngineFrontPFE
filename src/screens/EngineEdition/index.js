import { pathOrNull } from '@cdiscount/ramda-utils';
import { prop } from 'ramda';
import React, { useEffect, useState } from 'react';
import useReactRouter from 'use-react-router';

import clientApi from '../../client-api';
import EngineScreen from '../EngineScreen';
import parseEngine from './parse';
import useSaveEditEngine from './useSaveEditEngine';

function EngineEdition() {
  const [engine, setEngine] = useState({});
  const { match } = useReactRouter();

  const { saveEditEngine } = useSaveEditEngine();

  useEffect(() => {
    const id = pathOrNull(['params', 'id'], match);
    clientApi()
      .fetchEngine(id)
      .then(prop('data'))
      .then(rawEngine => {
        setEngine(parseEngine(rawEngine));
      });
  }, [match]);

  return <EngineScreen engine={engine} action={saveEditEngine} title="Ã©dition" setEngine={setEngine} />;
}

export default EngineEdition;
