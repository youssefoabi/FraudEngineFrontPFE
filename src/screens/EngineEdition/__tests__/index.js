import { itRendersAllMutations } from '@cdiscount/jest-utils';

import clientApi from '../../../client-api';
import EngineEdition from '../';

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

jest.mock('../../../client-api');

beforeEach(() => {
  clientApi.mockClear();
});

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
}));

jest.mock('use-react-router', () => () => ({ match: {} }));

describe(EngineEdition.name, () => {
  itRendersAllMutations(EngineEdition, [{ name: 'without props', props: {} }]);
});
