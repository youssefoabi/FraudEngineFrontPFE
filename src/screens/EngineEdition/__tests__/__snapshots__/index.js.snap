// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineEdition EngineEdition renders correctly without props 1`] = `
<EngineScreen
  action={[Function]}
  engine={Object {}}
  setEngine={[Function]}
  title="Ã©dition"
/>
`;
